src/modules/recipes/NewRecipeForm/index.jsx

```diff
@@ -0,0 +1,196 @@
+import { useForm, useFieldArray } from "react-hook-form";
+import { useRecipesDispatch } from "../RecipesProvider";
+import { RECIPE_ACTIONS } from "../RecipesProvider";
+import "./styles.css";
+
+export const NewRecipeForm = () => {
+  const dispatch = useRecipesDispatch();
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      strMeal: "",
+      strCategory: "Beef",
+      strArea: "Asian",
+      ingredients: [{ ingredient: "", measure: "", amount: "" }],
+      strInstructions: "",
+      strMealImage: "",
+    },
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "ingredients",
+  });
+
+  const onSubmit = (data) => {
+    const newRecipe = {
+      idMeal: Date.now().toString(),
+      strMeal: data.strMeal,
+      strCategory: data.strCategory,
+      strArea: data.strArea,
+      strInstructions: data.strInstructions,
+      strMealThumb: data.strMealImage,
+      ingredients: data.ingredients,
```

Comment:
you should not be adding ingredients like this to the store. In the original data, they look like this:
```
"strIngredient1": "penne rigate",
      "strIngredient2": "olive oil",
      "strIngredient3": "garlic",
      "strIngredient4": "chopped tomatoes",
      "strIngredient5": "red chilli flakes",
      "strIngredient6": "italian seasoning",
      "strIngredient7": "basil",
      "strIngredient8": "Parmigiano-Reggiano",
... 
      "strIngredient20": "",

```
and also you need measurements (strMeasure1 - strMeasure20).

So after you receive data from the form, split ingredients and measurements into different arrays and assign them to correct properties (strIngredient1, strIngredient2, etc.).
Only after ingredients and measurements are formatted correctly, you can add them to the context.